cumulus/parachains/runtimes/assets/common/src/lib.rs

```diff
@@ -134,6 +134,26 @@ pub type PoolAssetsConvertedConcreteId<PoolAssetsPalletLocation, Balance> =
 		TryConvertInto,
 	>;
 
+/// Returns an iterator of all assets in a pool with `asset`.
+///
+/// Should only be used in runtime APIs since it iterates over the whole
+/// `pallet_asset_conversion::Pools` map.
+pub fn get_assets_in_pool_with<
+	Runtime: pallet_asset_conversion::Config<PoolId = (xcm::v4::Location, xcm::v4::Location)>,
+>(
+	asset: &xcm::v4::Location,
```

Comment:
@franciscoaguirre could we make it more generic here? not to use concrete xcm version just latest?

just the idea, pseudo-code:
```suggestion
pub fn get_assets_in_pool_with<
	Runtime: pallet_asset_conversion::Config<PoolId = (L, L)>,
	L: TryInto<xcm::latest::prelude::Location> + Clone,	
>(
	asset: &L,
```